jtec-instrumentation/src/main/java/edu/tum/sse/jtec/instrumentation/coverage/CoverageMonitor.java

```diff
@@ -0,0 +1,44 @@
+package edu.tum.sse.jtec.instrumentation.coverage;
+
+import edu.tum.sse.jtec.util.IOUtils;
+import edu.tum.sse.jtec.util.ProcessUtils;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class CoverageMonitor {
+    private final CoverageMap coverageMap = new CoverageMap();
+
+    private CoverageMonitor() {
+    }
+
+    public static CoverageMonitor create() {
+        return new CoverageMonitor();
+    }
+
+    public void registerClass(final String className) {
+        registerCoverage(className);
+    }
+
+    public void registerMethodCall(final String className, final String methodSignature, final String returnType) {
+        registerCoverage(className + methodSignature + returnType);
+    }
+
+    private void registerCoverage(String value) {
+        String pid = ProcessUtils.getCurrentPid();
```

Comment:
see my comment above about extensibility to keeping track of coverage in non-forked mode; would we then change the key in the `CoverageMap` from pid to the test suite name, once a test suite finishes to run?

Reply:
Yeah, I think that should still work fairly easily. Since in that case we'd dump after every Test Suite anyway, we could also just include the test suite name in the dumping I guess.